### DISCUSSION TITLE
Advice needed: Lottie animations in Pixi.js

### DISCUSSION THREAD

**Discussion by m-i-wright** [pixijs/pixijs#8154] [Q&A]
Hello, I'm hoping for some advice on the best way to integrate Lottie animations into my Pixi app.

The Pixi app consists of a very large map that you can move with the mouse in all directions. Certain regions on the map are interactive, so you can click to enter a detailed view of that location. Each of these locations consists of several individual graphic layers, because graphics have to be exchanged or newly faded in during the course of the game. So the app already has a stately size, but the performance is still very good in all major browsers.

Now I want to enrich locations with Lottie animations, which will be produced by the design team using Adobe After Effects/Bodymovin. Since this can be quite a lot of animations, I am looking for a high performance way to render the Lottie animations between my display objects.

Unfortunately, the official `lottie-web` player does not support something like this out of the box. Third party tools like `pixi-after-effects` and `lottiefy` don't seem to be actively maintained anymore or have so many bugs that they are practically useless.

My first, naive attempt was therefore to let `lottie-web` render each animation into its own hidden canvas. Initially I create a texture from each canvas in order to create a `Sprite` and then call `texture.update()` on each frame of the animation.

```js
// Lottie enterFrame handler
if (location.visible) {
    if (!initialized) sprite.texture = Texture.from(lottieCanvas);
    sprite.texture.update();
    initialized = true;
}
```

Despite this ugly solution, performance is still good in Chrome. However, in Firefox it is really miserable. Do you see a better way how I could integrate Lottie animations? Any advice is welcome.

- **`pixi.js` version**:  6.0.4

**Comment by hrejsemnou** [comment]
Have you found a solution in the end?
